### PR TITLE
Provide a clear button for clearing directory root

### PR DESCRIPTION
**Pull request by darkwing**
Fixes Issue: #5448, #5432

A new button appears when the user has set a new directory root:

### CODE REVIEW

**Comment by darkwing** [comment]
This PR needs the following updates:

[ ] -  Implement @violasong's new mockup (icon here: https://design.firefox.com/icons/viewer/)
[ ] -  If the root has no sources (i.e. via a page change), we should clear the directory root

**Comment by jasonLaster** [comment]
@darkwing we might not need to clear the root, just not use it to filter the results

**Comment by darkwing** [comment]
screeny

**Comment by jasonLaster** [comment]
- home might be too dark
- home needs some pading
- not sure about that divider, maybe there's a simpler >
- the JS directory is pretty indented. Can we normalize indentation so the top directory is always the same

**Comment by darkwing** [comment]
Per your points:

- [ ] Yes, I agree the home icons should be lighter; I'll mess with opacity there.  @violasong said it would be a bigger, though.

- [ ] The padding issue is one I saw coming, since the tree padding is a bit eradic in the case of subdirs and files below them.  I was a bit worried about this. I added the `margin-inline-start: 10px;` per @violasong's single screenshot, but that only handled a single case.

- [ ] I kinda like the divider

- [ ] The only indentation I touched was the source-list indentation when it's inside a custom root.  What do you think without the 10px margin addition?  Or do we want to re-write all indentation within that bottom pane?  Sounds like....pain :D

- [ ] I'll look into the horizontal scrollbar -- just noticed it.

- [ ] The globe is currently incorrect without this PR; it's simply being added when depth is 0

**Comment by jasonLaster** [comment]
cool, re - tree depth is the tree preserving the depth some how which is affecting padding between js and js/views?

**Comment by darkwing** [comment]
It looks like `.tree-node[data-expandable="false"][aria-level="0"]` is the culprit for why the indent is so massive:

styletoggle

**Comment by darkwing** [comment]
The big blocker here is the scrollbar issue.

File: src/components/PrimaryPanes/SourcesTree.js [hunk 1]
```diff
@@ -193,7 +193,7 @@ class SourcesTree extends Component<Props, State> {
       return <Svg name="angular" />;
     }
 
-    if (depth === 0) {
+    if (depth === 0 && projectRoot === "") {
```
Comment (darkwing) [inline src/components/PrimaryPanes/SourcesTree.js]: Correcting the root / globe issue is actually this easy.

**Comment by darkwing** [comment]
This should be ready for another look.

**Comment by darkwing** [comment]
screeny2

  Looks good in RTL too.

**Pull request merged**